src/routes/notifications/notifications.controller.spec.ts

```diff
@@ -124,29 +158,47 @@ describe('Notifications Controller (Unit)', () => {
         .with('safeRegistration', safeRegistrations)
         .build();
       mockNetworkService.get.mockImplementation((url) => {
-        const getChainUrlPattern = `${safeConfigUrl}/api/v1/chains/`;
-        if (url.includes(getChainUrlPattern)) {
-          return Promise.resolve({ data: sample(chains) });
+        const targetChain = chains.find(
+          (chain) => url === `${safeConfigUrl}/api/v1/chains/${chain.chainId}`,
+        );
+        if (targetChain) {
+          return Promise.resolve({ data: targetChain });
```

Comment:
I think that this might be overcomplicating the tests a little bit. Doing it once to get a random number of chains is fine if that should be really be part of the test.

For most tests you can have a set number of chains (eg.: 2 or 3) and map the chain id directly with the expected payload.

You can leave a test that leaves some randomness to the amount of items in the array but I don't think it's necessary and makes the test harder to read.

Reply:
Please refer to [this comment](https://github.com/5afe/safe-client-gateway-nest/pull/267#discussion_r1098784500).